src/app/src/main/kotlin/org/imperial/mrc/hint/logging/LogElapsedTimeCallback.kt

```diff
@@ -0,0 +1,17 @@
+package org.imperial.mrc.hint.logging
+
+import java.time.Duration
+import java.time.Instant
+
+inline fun <T> logDurationOf(
+    callback: () -> T,
+    logger: GenericLogger,
+): T
+{
+    val start = Instant.now()
+    val response = callback()
+    val end = Instant.now()
+    val timeElapsed = Duration.between(start, end).toMillis()
+    logger.info("ADR request time elapsed: $timeElapsed")
```

Comment:
Feels like either the log message prefix should be a parameter too, or the method name should be less generic, to indicate that this should only be used for ADR calls e.g. `logADRRequestDuration`